sdk/python/feast/client.py

```diff
@@ -882,6 +893,46 @@ def _build_feature_references(
     return feature_ref_protos
 
 
+def _strip_project_online_response(
+    entity_rows: List[GetOnlineFeaturesRequest.EntityRow],
+    response: GetOnlineFeaturesResponse,
+) -> GetOnlineFeaturesResponse:
+    """
+    Strip the project part the string feature references returned in the given response.
+    Args:
+        entity_row: entity_rows passed when making the get features request
+        response: response obtained from a get online features request.
+    Returns:
+        The given response with projects stripped from string feature references
+    """
+
+    # collect entity row refs
+    entity_refs = set()
+    for entity_row in entity_rows:
+        entity_refs.update(entity_row.fields.keys())
+
+    # strip the project part the string feature references returned from serving
+    strip_field_values = []
+    for field_value in response.field_values:
+        strip_fields, strip_statuses = {}, {}
```

Comment:
another ~functional~ pythonic suggestion. highly subjective though. but I was trying to get rid off strip_ref_str & ref_str. it's just too many of them

```suggestion
strip = lambda ref: repr(FeatureRef.from_str(ref, ignore_project=True)) if ref not in entity_refs else ref
for field_value in response.field_values:
    fields_and_statuses = zip(sorted(field_value.fields.items()), 
                              sorted(field_value.statuses.items())
    fields_and_statuses = [(strip(key), (field, status)) for (key, field), (_, status) in fields_and_statuses]
    keys, fields, statuses = zip(*fields_and_statuses)
    GetOnlineFeaturesResponse.FieldValues(
                fields=dict(zip(keys, fields)), statuses=dict(zip(keys, statuses)),
            )
```

Reply:
Thanks for this. I implement a version of suggestion without a `sorted()` call.